### ISSUE TITLE
Rivets should not auto execute functions

### ISSUE THREAD

**Issue by jccazeaux** [mikeric/rivets#557]
In the fiddle https://jsfiddle.net/jccazeaux/pdfd7tb0/ you will see that the template
```html
{ model.date.getTime }
```
Is not updated when date is changed.
The reason is in the databinding update system. It compares old and new value of expression before updating. In our example, when the date changes, Rivets compares the function `getTime` between old and new date value. As `getTime` is a function of `Date` prototype, it doesn't change when date changes.

I see two solutions
1. Rivets executes function before comparing values
2. Rivets does'nt execute functions

I prefer second solution because
* It's more logic : if developper refers to a function without executing it, it will give reference to the function
* It opens the way of the ability to execute functions with arguments (see #554 )
* The undocumented `function` parameter of binders becomes obsolete. I don't like this parameter because you cannot always decide in binders declaration if value will be a function or not.

The code would be very simple, it's in `binding.coffee`
```Javascript
  # Sets the value for the binding. This Basically just runs the binding routine
  # with the suplied value formatted.
  set: (value) =>
    value = if value instanceof Function and !@binder.function
      @formattedValue value.call @model // <= Don't call !
    else
      @formattedValue value
```

**Comment by jccazeaux** [comment]
Well, in fact solution 2 alone won't solve the problem. The test will still be on prototype function reference. 
I think this is a problem because anyone encouters this glitch will loose some time to understand the problem.

If anyone has an idea or comment, it would be welcome

**Comment by Duder-onomy** [comment]
@jccazeaux Do you think forcing the dev to include the `()` could be a way to solve this? 

In your example, the `getTime` prototype method is called automatically, this will create confusion with the binding as you say. If we force the dev to include the `()` then we are saying, "this is a function, call it once but do not observe the function"...because observing a function makes no sense. This would also follow with your other ideas to include the keypath to any dependencies in the function invocation. Something like `Date.getTime(model.date)`.  Forcing the developer to essentially say : "this is a function AND I want to call it when its dependencies change " (this is very angular-ish BTW...but maybe a good idea? 

Personally, I have never run into this issue, we always tell our juniors to use formatters to accomplish this kind of thing.  Which could be another solution. If rivets see a `typeof === function` then it will throw an error which recommends using formatters to express keypaths which need to pass through functions.

**Comment by jccazeaux** [comment]
I don't think so. The problem is not the binding of the function. The problem is the fact that the function reference is always the same. 
```javascript
var date1 = new Date();
var date2 = new Date();
date1.getTime === date2.getTime(); // TRUE
```
Rivets (actually sightglass) tests value of function references before updating the view. 

See code here https://github.com/mikeric/sightglass/blob/master/index.js#L128

Even if `getTime` is not bound, it will still not be updated because of to this test.
This issue is complex. I ran in this issue once and it took time to understand why databinding didn't work.
Maybe the solution would be to skip the test in `sightglass` if the value is a function?

At the beginning I thought there was a link between auto-execution of functions and the issue I described. But in fact I think we have here two issues
* Test reference functions in sightglass can lead to weird behaviour in case of prototype functions
* Should rivets auto-execute functions?
 
I'll split the issue to separate discussions
